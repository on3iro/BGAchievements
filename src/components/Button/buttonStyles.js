/**
 * Creates button specific styles which can be applied to different elements/components
 *
 * @namespace Button.buttonStyles
 * @memberOf Button
  */

import { css } from 'styled-components'

export const getBackground = props => {
  let color = props.theme.primary
  let hover = props.theme.primaryHover

  if (props.warning) {
    color = props.theme.warning
    hover = props.theme.warningHover
  } else if (props.secondary) {
    color = props.theme.secondary
    hover = props.theme.secondaryHover
  }

  return {
    color,
    hover
  }
}

const buttonStyles = css`
  display: inline-block;
  margin: 0 0 15px 0;
  padding: 5px 10px;
  width: 100%;
  min-height: 35px;
  height: auto;

  text-decoration: none;
  background-color: ${props => getBackground(props).color};
  color: ${props => props.theme.noneWhite};

  text-align: center;

  outline: 0;
  border-radius: 1px;
  cursor: pointer;

  &:hover {
    background-color: ${props => getBackground(props).hover};
  }
`

export default buttonStyles
