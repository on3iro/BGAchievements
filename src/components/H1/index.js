/**
  * Renders a styled h1
  *
  * @namespace H1
  */

import styled from 'styled-components'

const H1 = styled.h1`
  font-size: 1.9rem;
  margin-top: 0;
  margin-bottom: 2%;

  color: ${props => props.theme.main};
`

export default H1
