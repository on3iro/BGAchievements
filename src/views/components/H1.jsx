// @flow

import styled from 'styled-components'

const H1 = styled.h1`
`

export {
  H1 as default
}
