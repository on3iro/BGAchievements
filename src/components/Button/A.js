/**
  * Renders an Anchor with button styles
  *
  * @namespace Button.A
  * @memberOf Button
  */

import styled from 'styled-components'

import buttonStyles from './buttonStyles'

const A = styled.a`
  ${buttonStyles}
`

export default A
