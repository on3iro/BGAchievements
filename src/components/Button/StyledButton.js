/**
  * Creates a button with styles applied
  *
  * @namespace Button.StyledButton
  * @memberOf Button
  */

import styled from 'styled-components'

import buttonStyles from './buttonStyles'

const StyledButton = styled.button`${buttonStyles}`

export default StyledButton
