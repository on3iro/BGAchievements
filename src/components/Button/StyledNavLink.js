import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { NavLink } from 'react-router-dom'

import buttonStyles from './buttonStyles'

import { surpressNoUsedVarsLinterError } from 'shared/helperFunctions'

export const NavLinkWrapper = props => {
  // pull off secondary prop to avoid react-unknown-property error
  const { secondary, warning, ...rest } = props
  surpressNoUsedVarsLinterError(secondary, warning)

  return <NavLink {...rest} />
}

NavLinkWrapper.propTypes = {
  secondary: PropTypes.bool,
  warning: PropTypes.bool
}

const StyledNavLink = styled(NavLinkWrapper)`${buttonStyles}`

export default StyledNavLink
