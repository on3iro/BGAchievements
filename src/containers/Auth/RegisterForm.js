/*
  * The user registration form
  *
  * @namespace RegisterPage.RegisterForm
  * @memberOf Register
  */

import React from 'react'
import PropTypes from 'prop-types'
import { Field, reduxForm } from 'redux-form'

import H1 from 'components/H1'
import Button from 'components/Button'
import LoadingSpinner from 'components/LoadingSpinner'

import FormField from 'components/FormField'

export const validate = values => {
  const errors = {}
  const REQUIRED = 'Pflichtfeld'

  if (!values.email) {
    errors.email = REQUIRED
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Ungültige Email-Adresse'
  }

  if (!values.password) {
    errors.password = REQUIRED
  }

  if (!values.name) {
    errors.name = REQUIRED
  }

  if (!values.confirm) {
    errors.confirm = REQUIRED
  } else if (values.confirm !== values.password) {
    errors.confirm = 'Passwörter stimmen nicht überein!'
  }

  return errors
}

export const RegisterForm = props => {
  return (
    <form onSubmit={props.handleSubmit}>
      <H1>Registrierung</H1>
      <Field name='email' type='text'
        component={FormField} label='Email'
      />
      <Field name='password' type='password'
        component={FormField} label='Passwort'
      />
      <Field name='confirm' type='password'
        component={FormField} label='Passwort wiederholen'
      />
      {
        props.isLoading
          ? <Button onClick={(e) => e.preventDefault}>
            <LoadingSpinner controls />
          </Button>
          : <Button submit>Jetzt kostenlos registrieren</Button>
      }
    </form>
  )
}

RegisterForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  isLoading: PropTypes.bool
}

export default reduxForm({
  form: 'register',
  validate
})(RegisterForm)
