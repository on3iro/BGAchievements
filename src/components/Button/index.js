/**
 * A common button, if you pass it a prop "route" it'll render a link to a react-router route.
 * Otherwise it'll render a link with an onclick.
 *
 * @namespace Button
  */

import React, { Children } from 'react'
import PropTypes from 'prop-types'

import A from './A'
import StyledSubmit from './StyledSubmit'
import StyledNavLink from './StyledNavLink'

const Button = props => {
  // Render an anchor tag
  let button = (
    <A href={props.href} onClick={props.onClick} {...props} >
      {Children.toArray(props.children)}
    </A>
  )

  // If the button has a 'to' prop, we want to render a NavLink
  if (props.to) {
    button = (
      <StyledNavLink {...props} >
        {Children.toArray(props.children)}
      </StyledNavLink>
    )
  }

  // If the button has submit prop render input
  if (props.submit) {
    button = (
      <StyledSubmit {...props} />
    )
  }

  return button
}

Button.propTypes = {
  handleRoute: PropTypes.func,
  href: PropTypes.string,
  onClick: PropTypes.func,
  children: PropTypes.node,
  submit: PropTypes.bool
}

export default Button
