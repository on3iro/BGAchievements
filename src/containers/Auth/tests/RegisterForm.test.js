import React from 'react'
import { shallow } from 'enzyme'
import { Field } from 'redux-form'

import H1 from 'components/H1'
import Button from 'components/Button'

import { RegisterForm, validate } from '../RegisterForm'

const renderComponent = newProps => {
  const props = {
    handleSubmit: jest.fn()
  }

  const renderedWrapper = shallow(<RegisterForm {...props} />)

  return {
    props,
    renderedWrapper
  }
}

describe('<RegisterForm />', () => {
  it('should render <form>', () => {
    const { renderedWrapper } = renderComponent()
    expect(renderedWrapper.find('form').length).toMatchSnapshot()
  })

  it('should render <H1>', () => {
    const { renderedWrapper } = renderComponent()
    expect(renderedWrapper.find(H1).length).toMatchSnapshot()
  })

  it('should render Fields', () => {
    const { renderedWrapper } = renderComponent()
    expect(renderedWrapper.find(Field).length).toMatchSnapshot()
  })

  it('Should render <Button />', () => {
    const { renderedWrapper } = renderComponent()
    expect(renderedWrapper.find(Button).length).toMatchSnapshot()
  })
})

describe('validate()', () => {
  const values = {
    email: 'foo@bar.com',
    name: 'foo',
    password: '123',
    confirm: '123'
  }

  it('should throw "Required" on empty email', () => {
    expect(validate({ ...values, email: ''})).toMatchSnapshot()
  })

  it('should throw "Invald email address"', () => {
    expect(validate({ ...values, email: 'asdfasdf'})).toMatchSnapshot()
  })

  it('should throw "Required" on empty name', () => {
    expect(validate({ ...values, name: ''})).toMatchSnapshot()
  })

  it('should throw "Required" on password', () => {
    expect(validate({ ...values, password: ''})).toMatchSnapshot()
  })

  it('should throw "Required" on confirm', () => {
    expect(validate({ ...values, confirm: ''})).toMatchSnapshot()
  })

  it('should throw "Passwords do not match"', () => {
    expect(validate({ ...values, confirm: 'abc'})).toMatchSnapshot()
  })
})
