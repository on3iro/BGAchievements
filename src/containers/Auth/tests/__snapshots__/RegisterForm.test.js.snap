exports[`<RegisterForm /> Should render <Button /> 1`] = `1`

exports[`<RegisterForm /> should render <H1> 1`] = `1`

exports[`<RegisterForm /> should render <form> 1`] = `1`

exports[`<RegisterForm /> should render Fields 1`] = `3`

exports[`validate() should throw "Invald email address" 1`] = `
Object {
  "email": "Ungültige Email-Adresse",
}
`

exports[`validate() should throw "Passwords do not match" 1`] = `
Object {
  "confirm": "Passwörter stimmen nicht überein!",
}
`

exports[`validate() should throw "Required" on confirm 1`] = `
Object {
  "confirm": "Pflichtfeld",
}
`

exports[`validate() should throw "Required" on empty email 1`] = `
Object {
  "email": "Pflichtfeld",
}
`

exports[`validate() should throw "Required" on empty name 1`] = `
Object {
  "name": "Pflichtfeld",
}
`

exports[`validate() should throw "Required" on password 1`] = `
Object {
  "confirm": "Passwörter stimmen nicht überein!",
  "password": "Pflichtfeld",
}
`
